kratos/python/add_variable_utils_to_python.cpp

```diff
@@ -452,6 +452,63 @@ void AddVariableUtilsToPython(pybind11::module &m)
         .def("ApplyFixity", ApplyFixity<Variable<double>>)
         .def("ApplyFixity", ApplyFlaggedFixity<Variable<double>>)
         .def("ApplyVector", &VariableUtils::ApplyVector<Variable<double>>)
+        .def("ApplyVector", []( VariableUtils& rVariableUtils, 
```

Comment:
ok, i was unaware. Still i would like a version with provided dimension to get only the x y component (or xyz in 3D)

Reply:
mmm, as i see there is no function to get the original or current coordinates, is that correct?

also the function  GetVectorData and GetScalarData are not exported to python, and even if they were they do not allow specifying the step, so in any case i would need to have a version of those. Am i correct?

One option would be to modify the current header to 

template<class TDataType>
void GetVectorData(
        const Variable<TDataType>& rVariable,
        const DataLocation DataLoc,
        std::vector<double>& data,
        int TSize=-1, //impossible number, if we later detect it is -1, we would use automatic detection
        int Step=0 //only used for historical data...
) const

as an aside, why returing a std::vector? it would be better to either template the return type or to default to Vector which is faster...

Reply:
- yes you are right, there are no functions for the coordinates, hence adding them makes sense to me
- Ok I didn't realize they are that different. I leave it up to you if you want to either extend the existing fcts or add new ones. If you add new ones, can you please add them properly, not in `add_to_python_...`, including a test?
- I choose `std::vector` since it also supports non-`double` types (and bcs the interface of CoSimIO uses it :wink:)

Reply:
to add to my previous comment, the function is model_oart_utils onnly works on modelpart, while it is very useful to be able to pass a list of nodes disconnected from a modelpart.

summing up all of these reasons i think i 'll add the function to variable_utils as i did (incorporating the suggestions of @loumalouomega and @miguelmaso )

as a final comment i wanted to report that, as discussed privetely with @philbucher it is generally not a good idea using std::vector in performance sensitive parts of the code, since the std vector is default initialized to zero (when the default allocator is used). A possible solution for this is to use boost's noinit_adaptor

https://www.boost.org/doc/libs/master/libs/core/doc/html/core/noinit_adaptor.html#:~:text=The%20header%20%3Cboost%2Fcore%2Fnoinit_adaptor.hpp%3E%20provides%20the%20class%20template%20boost%3A%3Anoinit_adaptor,new%2C%20and%20whose%20destroy%28ptr%29%20invokes%20value_type%20destructor%20directly.

in any case this is a separate discussion and i am reporting it here just to leave trace of what i consider to be an important  aspect